tests/examples/test_download_files.py

```diff
@@ -0,0 +1,895 @@
+"""
+Test downloading files.
+
+Enable these tests with:
+
+pytest --test_downloads
+
+"""
+import os
+import warnings
+
+import numpy as np
+import pytest
+
+import pyvista as pv
+from pyvista import examples
+
+if 'TEST_DOWNLOADS' in os.environ:
+    warnings.warn('"TEST_DOWNLOADS" has been deprecated. Use `pytest --test_downloads`')
+
+pytestmark = pytest.mark.needs_download
```

Comment:
Doesn't this need to decorate all these functions? or how is this working?

Reply:
Okay this is good to know - TIL

But is it a problem that we do this here:

https://github.com/pyvista/pyvista/blob/7d65054dff4717338519e73d287c7709caef1b20/tests/utilities/test_reader.py#L12-L14

Are we then skipping this whole module on Mac? Also the fact that it says "Azure" there probably means this is super old and needs to be cleaned up

Reply:
Looks like it, yup.

Reply:
Leaving this for another issue/PR: https://github.com/pyvista/pyvista/issues/3349#issue-1381158342